File: cpp/1605. 给定行和列的和求可行矩阵.cpp
class Solution {
public:
    vector<vector<int>> restoreMatrix(vector<int>& rowSum, vector<int>& colSum) {
        int r=rowSum.size(),c=colSum.size();
        vector<vector<int> > grid(r,vector<int>(c,0));
        int i=0,j=0;
        while(i<r&&j<c){
                int x=min(rowSum[i],colSum[j]);
                grid[i][j]=x;
                rowSum[i]-=x;
                colSum[j]-=x;
                if(rowSum[i]==0){
                    i++;
                }
                if(colSum[j]==0){
                    j++;
                }
        }
        return grid;
    }
};

File: 刷题/golang/1605. 给定行和列的和求可行矩阵.go
func restoreMatrix(rowSum []int, colSum []int) [][]int {
    r, c := len(rowSum), len(colSum)
    gird := make([][]int, r)
    for i := range gird {
        gird[i] = make([]int, c)
    }
    i, j := 0, 0
    for i<r && j<c {
        x := min(rowSum[i], colSum[j])
        gird[i][j] = x
        rowSum[i] -= x
        colSum[j] -= x
        if rowSum[i] == 0 {
            i++
        }
        if colSum[j] == 0{
            j++
        }
    }
    return gird
}
func min(x, y int) int {
    if x<y {
        return x
    }
    return y
}